### PR TITLE
feat(ci): Add basic GitHub Action with build

### LINKED ISSUE #27: Add CI for the contrib-updater CLI tool
The updater tool currently doesn't have any tests or continuous integration running on pull requests. We should at least ensure that the tool compiles successfully, and we can follow up with more sophisticated CI in the future. 

This could be done with a fairly standard `setup-purescript` configuration [following that project's README](github.com/purescript-contrib/setup-purescript).

### PR DESCRIPTION
**Pull request by gillchristian**
Closes #27

### CODE REVIEW

**Comment by gillchristian** [comment]
Could it be that it will not ~run~ add the action because I'm not a contributor to the project?

(cc) @thomashoneyman 

UPDATE: I guess it's because PR is from a fork. But it did run on the fork and failed (https://github.com/gillchristian/governance/actions/runs/271315517). I'll take a look tomorrow

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+      - '*'
```
Comment (gillchristian) [inline .github/workflows/ci.yml]: [Q] Should it run on every branch / PR ?

Comment (thomashoneyman) [reply .github/workflows/ci.yml]: We can do this with

```
on: [push, pull_request]
  push:
    branches: [main]
  pull_request:
    branches: '*'
```

ie. any pull request (which includes any push to a branch in an open pull request), and any push to main, but not pushes to branches that aren't main and aren't in a pull request.

**Comment by thomashoneyman** [comment]
@gillchristian I believe you have a syntax error -- the `pull_request` key needs to be followed by `branches: ` and/or `tags: ` but you've gone straight to the wildcard pattern.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (thomashoneyman) [reply .github/workflows/ci.yml]: Sorry, I typod, you wouldn't have both `[ push, pull_request ]` and the options below. Should be this:

```
on:
  push:
    branches: [main]
  pull_request:
    branches: '*'
```

**Comment by gillchristian** [comment]
Should be working now https://github.com/gillchristian/governance/runs/1164216854 (pushed to `main` of my fork so it would run there)

**Review by thomashoneyman — APPROVED** [review]

**Comment by thomashoneyman** [comment]
Thanks!

**Pull request merged**